### ISSUE TITLE
Automatically upgrade rosetta2 users from `x86_64` to `arm64` builds

### ISSUE THREAD

**Issue by fmoc** [owncloud/client#10757]
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Feature description

We should automatically upgrade rosetta2 users on M1/M2 to the native `arm64` builds we now provide.

### Proposed solution

See https://developer.apple.com/forums/thread/653009. Shouldn't be a problem to add that to our updater.

### Alternative solutions you considered

Tell users to switch to `arm64` manually if they want to benefit from the native build's advantages (like generally higher efficiency, less battery consumption, ...).

**Comment by fmoc** [comment]
First half implemented in https://github.com/owncloud/client/pull/11376, please review. The second half will follow in a PR to the updater server.

**Comment by fmoc** [comment]
The update server already behaves like we need it to. It just uses `currentArch` to determine which binary to ship (if available, otherwise falls back to `buildArch`). As we send both parameters anyway, it's working as expected in combination with #11376.

**Comment by fmoc** [comment]
Test instructions: test upgrade against playground server.

Note: blocked by another issue.

**Comment by fmoc** [comment]
See also https://github.com/owncloud/client/issues/11388.

**Comment by HanaGemela** [comment]
Automated updates don't work at the moment
